### PR TITLE
Allow psql client certificate authentication

### PR DESCRIPTION
**Pull request by Ongy**
## Proposed change

Add support for more postgres connection options to enable client certificate authentication.
To test thix you need a a PKI and postgres server setup for certificate based auth. 

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by Ongy** [comment]
For now, please provide quick feedback if this is in scope. Then I'll look int oupdating the docs as well.

I tested this in my setup, it works :)

**Comment by stumpylog** [comment]
it seems fine to include this for users.  Please do update the docs for setup if a user desires.

**Comment by Ongy** [comment]
I don't think there's much more to document. While I could add a guide, I don't think it's particularily useful.
This is a rather advanced feature that people should find if they have a psql server setup this way.

File: docs/configuration.md [hunk 1]
```diff
@@ -86,6 +86,31 @@ changed here.
 
     Default is `prefer`.
 
+`PAPERLESS_DBSSLROOTCERT=<ca-path>`
```
Comment (stumpylog) [inline docs/configuration.md:89]: Each of these should note the default is unset (for example as `PAPERLESS_DB_TIMEOUT` does), which disables usage of certs.

**Review by stumpylog — APPROVED** [review]
Didn't affect my connection without a cert, so seems to work as expected

**Pull request merged**